cluster/assemblestate/devices.go

```diff
@@ -0,0 +1,290 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2025 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package assemblestate
+
+import (
+	"time"
+)
+
+// DeviceQueryTracker manages device identity information and query/response
+// lifecycle for unknown devices in the cluster.
+type DeviceQueryTracker struct {
+	// pendingResponses is used to send notifications that indicate that we are
+	// able to respond to a peer's query for device information.
```

Comment:
I think if this were rephrased a bit more like the corresponding method doc comment it would be a bit clearer